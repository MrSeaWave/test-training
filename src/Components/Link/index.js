import React, { Component } from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

class Link extends Component {
  constructor(props) {
    super(props);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onMouseEnter = () => {
    this.setState({ class: STATUS.HOVERED });
  };

  _onMouseLeave = () => {
    this.setState({ class: STATUS.NORMAL });
  };

  render() {
    const { page, children } = this.props;
    return (
      <a
        className={this.state.class}
        href={page || '#'}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        {children}
      </a>
    );
  }
}

export default Link;
