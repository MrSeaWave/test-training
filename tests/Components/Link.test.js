import React from 'react';
import TestRenderer from 'react-test-renderer';
import Link from '../../src/Components/Link';

it('当鼠标悬浮时改变class数据', () => {
  const component = TestRenderer.create(
    <Link page="http://www.facebook.com"> ------- Facebook ------- </Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
