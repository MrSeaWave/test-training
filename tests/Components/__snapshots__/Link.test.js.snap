// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`当鼠标悬浮时改变class数据 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   ------- Facebook ------- 
</a>
`;

exports[`当鼠标悬浮时改变class数据 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   ------- Facebook ------- 
</a>
`;

exports[`当鼠标悬浮时改变class数据 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   ------- Facebook ------- 
</a>
`;
